File: LeetCode/1319-unique-number-of-occurrences/unique-number-of-occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        int n = arr.size();
        unordered_map<int, int> freq;
        vector<int> buckets (n + 1, -1);

        //populate freq map
        for (int val : arr) freq[val]++;
        //check for conflicting freqs
        for (auto [v, f] : freq) {
            if (buckets[f] != -1) return false;
            else buckets[f] = v;
        } //for

        return true;
    } //uniqueOccurrences
};

File: LeetCode/1319-unique-number-of-occurrences/unique-number-of-occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        freq = {}

        for val in arr:
            if val not in freq:
                freq[val] = 1
            else:
                freq[val] += 1
        
        return len(set(freq.values())) == len(freq.values())